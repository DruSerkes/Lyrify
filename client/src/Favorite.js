import React from 'react';
import { Link } from 'react-router-dom';
import ListItem from '@material-ui/core/ListItem';

const Favorite = ({ artist, song, id }) => {
	return (
		<Link to={`/favorites/${id}`} className="Favorite">
			<ListItem id={id} divider>
				{artist}: "{song}"
			</ListItem>
		</Link>
	);
};

export default Favorite;
