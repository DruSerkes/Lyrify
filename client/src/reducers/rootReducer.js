import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
