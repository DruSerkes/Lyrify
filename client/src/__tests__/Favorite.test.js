import React from 'react';
import { render } from '@testing-library/react';
import Favorite from '../Favorite';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<Favorite />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Favorite />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});

