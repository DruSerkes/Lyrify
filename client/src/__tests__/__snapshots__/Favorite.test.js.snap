// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root Favorite MuiListItem-gutters MuiListItem-divider"
  >
    <a
      class="Favorite-Item"
      href="/favorites/undefined"
    >
      : ""
    </a>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </li>
</DocumentFragment>
`;
